adapters/wagmi-connector/src/connector.ts

```diff
@@ -29,201 +22,143 @@ type BloctoOptions = {
+        provider.chainId ??
+        (await provider?.request({ method: 'eth_chainId' }));
+      return normalizeChainId(chainId);
+    },
+    async getProvider({ chainId } = {}) {
+      if (!walletProvider) {
+        const store = await config.storage?.getItem('store');
+        const lastConnectedChainId = store?.state?.chainId;
+        const desiredChainId = chainId ?? lastConnectedChainId;
+        const ethereum: BloctoEthereumProviderParameters = {
+          chainId: desiredChainId,
+          rpc: config.chains.find((x) => x.id === desiredChainId)?.rpcUrls
+            .default.http[0],
+        };
+
+        walletProvider = new BloctoSDK({ ethereum, appId })?.ethereum;
+
+        if (!walletProvider) {
+          throw new Error('Blocto SDK is not initialized.');
+        }
+
+        walletProvider.on('accountsChanged', this.onAccountsChanged.bind(this));
+        walletProvider.on('chainChanged', this.onChainChanged.bind(this));
+        walletProvider.on('disconnect', this.onDisconnect.bind(this));
+      }
 
-      if (!chain || !isBloctoSupportChain) {
-        throw new SwitchChainError(
-          new Error(`Blocto unsupported chain: ${id}`)
+      return Promise.resolve(walletProvider);
+    },
+    async isAuthorized() {
+      const recentConnectorId = await config.storage?.getItem(
+        'recentConnectorId'
+      );
+      if (recentConnectorId !== this.id) return false;
+
+      const accounts = await this.getAccounts();
+      return !!accounts.length;
+    },
+    async switchChain({ chainId }) {
+      try {
+        const provider = await this.getProvider();
+        const id = numberToHex(chainId);
+        const chain = config.chains.find((x) => x.id === chainId);
+        const networks = await provider.supportChainList();
+        const evmSupportMap = networks.reduce(
+          (a: any, v: any) => ({ ...a, [v.chainId]: v }),
+          {}
         );
+        const isBloctoSupportChain = evmSupportMap[`${chainId}`];
+
+        if (!chain || !isBloctoSupportChain) {
+          throw new SwitchChainError(
+            new Error(`Blocto unsupported chain: ${id}`)
+          );
+        }
+
+        await provider.request({
+          method: 'wallet_addEthereumChain',
+          params: [{ chainId: id, rpcUrls: chain?.rpcUrls.default.http }],
+        });
+        await provider.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: id }],
+        });
+
+        return chain;
+      } catch (err) {
+        const error = err as RpcError;
+        if (error.code === UserRejectedRequestError.code)
+          throw new UserRejectedRequestError(error);
+
+        throw new SwitchChainError(error as Error);
       }
-
-      await provider.request({
-        method: 'wallet_addEthereumChain',
-        params: [{ chainId: id, rpcUrls: chain?.rpcUrls.default.http }],
+    },
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onAccountsChanged() {},
```

Comment:
blocto-sdk supported change account, maybe can do something?

Reply:
since our switch account only happens when a user re-connect the blocto,
and this behavior [will execute `connect` function and return the latest account & chainId,](https://github.com/blocto/blocto-sdk/blob/499f6147f7d7f3affc1be5e8a3f420ff444cc8bf/adapters/wagmi-connector/src/connector.ts#L56)
so it's fine that we didn't handle `onAccountsChanged`.